src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIChatClient.cs

```diff
@@ -569,13 +568,16 @@ private sealed class OpenAIChatToolJson
     private IEnumerable<OpenAI.Chat.ChatMessage> ToOpenAIChatMessages(IEnumerable<ChatMessage> inputs)
     {
         // Maps all of the M.E.AI types to the corresponding OpenAI types.
-        // Unrecognized content is ignored.
+        // Unrecognized or non-processable content is ignored.
 
         foreach (ChatMessage input in inputs)
         {
-            if (input.Role == ChatRole.System)
+            if (input.Role == ChatRole.System || input.Role == ChatRole.User)
```

Comment:
```suggestion
            if (input.Role is ChatRole.System or ChatRole.User)
```

Reply:
That won't work, unfortunately... these aren't enums.